### ISSUE TITLE
[question] Handling redundant settings due to `cmaketoolchain:user_toolchain`

### ISSUE THREAD

**Issue by sykhro** [conan-io/conan#15582]
### What is your question?

I have a profile for Release-mode clang-cl:
```
[settings]
os=Windows
arch=x86_64
build_type=Release

[...]
```

the profile uses a CMake user toolchain and also specifies C(XX)FLAGS for projects that use other build systems:
```
tools.cmake.cmaketoolchain:user_toolchain = {{ [llvm_dir ~ "/windows-x64-clang.cmake"] }}

tools.build:cflags = ["-fuse-ld=lld", "-flto=thin", "/EHsc", "/MT"]
tools.build:cxxflags = ["-fuse-ld=lld", "-flto=thin", "/EHsc", "/MT"]
tools.build:exelinkflags = ["/MANIFEST:NO", "/NODEFAULTLIB:libucrt.lib", "/DEFAULTLIB:ucrt.lib"]
tools.build:sharedlinkflags = ["/MANIFEST:NO", "/NODEFAULTLIB:libucrt.lib", "/DEFAULTLIB:ucrt.lib"]
```

This, however, causes build failures with CMake.
The user toolchain file sets the link flags as follows:
```
string(APPEND CMAKE_EXE_LINKER_FLAGS_INIT_DEBUG " /NODEFAULTLIB:libucrtd.lib /DEFAULTLIB:ucrtd.lib")
string(APPEND CMAKE_EXE_LINKER_FLAGS_INIT_RELEASE " /NODEFAULTLIB:libucrt.lib /DEFAULTLIB:crtd.lib")
```
while conan generates the following code for `conan_toolchain.cmake`:
```
string(APPEND CONAN_SHARED_LINKER_FLAGS " /MANIFEST:NO")
string(APPEND CONAN_EXE_LINKER_FLAGS " /MANIFEST:NO")
```
but during the compiler checks, CMake defaults to `Debug` mode (I'm using a multi-config generator).
So, the linker line becomes a mix of Debug and Release flags.

My question is: how can I disable the CMakeToolchain `cmake_flags_init` block directly from the profile? 

### Have you read the CONTRIBUTING guide?

- [X] I've read the CONTRIBUTING guide

**Comment by memsharded** [comment]
Hi @sykhro 

Thanks for your detailed report.

This is a known issue, I think it is the same as https://github.com/conan-io/conan/issues/13920. I have been trying to address it for a while, but so far we haven't got enough time to do it, even if I started the effort and I have something in a branch.

I'll try to give it another push, thanks for raising the isue!

**Comment by sykhro** [comment]
Yes, it's indeed the same issue, I had trouble wording it. I'd be happy to contribute if you have any pointers

**Comment by memsharded** [comment]
Thanks for checking. I think we can close this ticket and follow up on the other one.
As I had a branch somewhere to attempt this, let me have a look first, I will let you know if possible to contribute, thanks for the offer!

**Comment by sykhro** [comment]
To be more precise, the root cause is the same but it would probably still be nice to be able to toggle which CMakeToolchain blocks to enable/disable from the profile itself, so that the repetition of variables can be avoided completely

**Comment by memsharded** [comment]
I am submitting https://github.com/conan-io/conan/pull/15654 to fix this, hopefully for next 2.1, feedback welcome! (the PR might close this ticket when merged)

The feature to enable/disable individual blocks is a bit more unlikely, it would need a separate, dedicated issue to discuss it.

**Comment by memsharded** [comment]
Merged https://github.com/conan-io/conan/pull/15654 for next 2.1
